tests/integration.spec.ts

```diff
@@ -81,11 +91,11 @@ for (const packageManager of ['npm', 'yarn'] as ('npm' | 'yarn')[]) {
       expect(fs.existsSync(path.join(dir, 'playwright.config.ts'))).toBeTruthy();
 
       {
-        const { code } = await exec(packageManager === 'npm' ? 'npx' : 'yarn', ['playwright', 'install-deps']);
+        const { code } = await exec(packageManager === 'npm' ? 'npx' : packageManager === 'pnpm' ? 'pnpm dlx' : 'yarn', ['playwright', 'install-deps']);
```

Comment:
> packageManager === 'npm' ? 'npx' : packageManager === 'pnpm' ? 'pnpm dlx' : 'yarn'

This is getting a bit out of control with the nested ternary operators. How do you feel about using proper traits for the package manager? You went half of the way with `pmInitializeCommand` and friends, but you could take it all the way and have `packageManager.initializeCommand()`, `packageManager.installCommand()`, `packageManager.npxCommand()`, etc. It'll end up easier to read and might be less lines too.